vapi/simulator/simulator.go

```diff
@@ -407,9 +394,37 @@ func (s *handler) association(w http.ResponseWriter, r *http.Request) {
 			}
 		}
 		s.ok(w, ids)
+	}
+}
+
+func (s *handler) associationID(w http.ResponseWriter, r *http.Request) {
```

Comment:
There is different behaviour between vcsim and VAPI REST here. vcsim will accept a valid TAGURN after `/tag-association/TAGURN` whereas VAPI REST requires the `/id:{TAGURN}` path. This leads to inconsistent behaviour when testing tagging.

Example against vcsim:

```bash
curl -i -d '{
        "object_id": {
                "id": "vm-53",
                "type": "VirtualMachine"
        }
}' -k -H "Content-Type: application/json" "http://localhost:8989/rest/com/vmware/cis/tagging/tag-association/urn:vmomi:InventoryServiceTag:b706f50e-00bf-46a2-963a-31e1328265ce:GLOBAL?~action=detach"
HTTP/1.1 200 OK
Date: Fri, 25 Jan 2019 10:01:15 GMT
Content-Length: 0
```

Same against VC:

```bash
curl -k -i -d '{
        "object_id": {
                "id": "vm-1602",
                "type": "VirtualMachine"
        }
}' -k -H "vmware-api-session-id:bfb939fe4df9e86d4458e47f104b0e16" -H "Content-Type: application/json" "https://192.168.178.72/rest/com/vmware/cis/tagging/tag-association/urn:vmomi:InventoryServiceTag:604e66d9-97d4-414c-82f5-d25475320087:GLOBAL?~action=detach"
HTTP/1.1 404 Not Found
Date: Fri, 25 Jan 2019 10:09:17 GMT
Content-Type: application/json;charset=UTF-8
Transfer-Encoding: chunked

{"name":"com.vmware.vapi.rest.httpNotFound","localizableMessages":[{"defaultMessage":"Not found.","id":"com.vmware.vapi.rest.httpNotFound"}],"majorErrorCode":404}%
```

Reply:
Thanks @embano1 , I'll update this PR in a bit to support `id:` prefix.